src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -0,0 +1,95 @@
+{
+    [ServiceLocator(Default = typeof(NodeHandler))]
+    public interface INodeHandler : IHandler
+    {
+        NodeHandlerData Data { get; set; }
+    }
+
+    public sealed class NodeHandler : Handler, INodeHandler
+    {
+        private readonly object _dataReceivedLock = new object();
+
+        public NodeHandlerData Data { get; set; }
+
+        public async Task RunAsync()
+        {
+            // Validate args.
+            Trace.Entering();
+            ArgUtil.NotNull(Data, nameof(Data));
+            ArgUtil.NotNull(ExecutionContext, nameof(ExecutionContext));
+            ArgUtil.NotNull(Inputs, nameof(Inputs));
+            ArgUtil.NotNull(TaskDirectory, nameof(TaskDirectory));
+
+            // Update the env dictionary.
+            AddInputsToEnvironment();
+            AddEndpointsToEnvironment();
+            AddVariablesToEnvironment();
+
+            // Resolve the target script.
+            string target = Data.Target;
+            ArgUtil.NotNullOrEmpty(target, nameof(target));
+            target = Path.Combine(TaskDirectory, target);
+            if (!File.Exists(target))
+            {
+                throw new Exception("TODO: task script file not found message.");
+            }
+
+            // Resolve the working directory.
+            string workingDirectory = Data.WorkingDirectory;
+            if (string.IsNullOrEmpty(workingDirectory))
+            {
+                workingDirectory = TaskDirectory;
+            }
+            else if (!Directory.Exists(workingDirectory))
+            {
+                throw new Exception("TODO: task working directory not exist message");
+            }
+
+            // Setup the process invoker.
+            using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+            {
+                object outputLock = new object();
+                processInvoker.OutputDataReceived += OnDataReceived;
+                processInvoker.ErrorDataReceived += OnDataReceived;
+                // TODO: Get path to node.exe. Is it always well-known? Or should WHICH be preferred?
+#if OS_WINDOWS
+                string fileName = "node.exe";
+#else
+                string fileName = "node";
+#endif
+
+                // Format the arguments passed to nodejs.
+                // 1) Wrap the script file path in double quotes.
+                // 2) Escape double quotes within the script file path.
+                string arguments = StringUtil.Format(@"""{0}""", target.Replace(@"""", @"\"""));
+                int exitCode = await processInvoker.ExecuteAsync(
+                    workingDirectory: workingDirectory,
+                    fileName: fileName,
+                    arguments: arguments,
+                    environment: Environment,
+                    cancellationToken: ExecutionContext.CancellationToken);
+                if (exitCode != 0)
+                {
+                    throw new Exception("TODO: BETTER ERROR MESSAGE");
```

Comment:
There is no need to explicitly set the task result to success/fail. Just bubble out if there is an error. The steps runner will handle catching the error and failing the execution context.

When the execution context is completed (and underlying timeline is completed), the result will be defaulted to success if not already set.